kusto/kql/value.go

```diff
@@ -0,0 +1,133 @@
+	return &kqlValue{
+		value:     value,
+		kustoType: kustoType,
+	}
+}
+
+func (b *statementBuilder) AddBool(value bool) Builder {
+	return b.addBase(newValue(value, types.Bool))
+}
+
+func (b *statementBuilder) AddDateTime(value time.Time) Builder {
+	return b.addBase(newValue(value, types.DateTime))
+}
+
+func (b *statementBuilder) AddDynamic(value interface{}) Builder {
+	return b.addBase(newValue(value, types.Dynamic))
+}
+
+func (b *statementBuilder) AddGUID(value uuid.UUID) Builder {
+	return b.addBase(newValue(value, types.GUID))
+}
+
+func (b *statementBuilder) AddInt(value int32) Builder {
+	return b.addBase(newValue(value, types.Int))
+}
+
+func (b *statementBuilder) AddLong(value int64) Builder {
+	return b.addBase(newValue(value, types.Long))
+}
+
+func (b *statementBuilder) AddReal(value float64) Builder {
+	return b.addBase(newValue(value, types.Real))
+}
+
+func (b *statementBuilder) AddString(value string) Builder {
+	return b.addBase(newValue(value, types.String))
+}
+
+func (b *statementBuilder) AddTimespan(value time.Duration) Builder {
+	return b.addBase(newValue(value, types.Timespan))
+}
+
+func (b *statementBuilder) AddDecimal(value value.Decimal) Builder {
+	return b.addBase(newValue(value, types.Decimal))
+}
+
+func (q *StatementQueryParameters) AddBool(key string, value bool) *StatementQueryParameters {
+	return q.addBase(key, string(types.Bool), newValue(value, types.Bool))
+}
+
+func (q *StatementQueryParameters) AddDateTime(key string, value time.Time) *StatementQueryParameters {
+	return q.addBase(key, string(types.DateTime), newValue(value, types.DateTime))
+}
+
+func (q *StatementQueryParameters) AddDynamic(key string, value interface{}) *StatementQueryParameters {
+	return q.addBase(key, string(types.Dynamic), newValue(value, types.Dynamic))
+}
+
+func (q *StatementQueryParameters) AddGUID(key string, value uuid.UUID) *StatementQueryParameters {
+	return q.addBase(key, string(types.GUID), newValue(value, types.GUID))
+}
+
+func (q *StatementQueryParameters) AddInt(key string, value int32) *StatementQueryParameters {
+	return q.addBase(key, string(types.Int), newValue(value, types.Int))
+}
+
+func (q *StatementQueryParameters) AddLong(key string, value int64) *StatementQueryParameters {
+	return q.addBase(key, string(types.Long), newValue(value, types.Long))
+}
+
+func (q *StatementQueryParameters) AddReal(key string, value float64) *StatementQueryParameters {
+	return q.addBase(key, string(types.Real), newValue(value, types.Real))
+}
+
+func (q *StatementQueryParameters) AddString(key string, value string) *StatementQueryParameters {
+	return q.addBase(key, string(types.String), newValue(value, types.String))
+}
+
+func (q *StatementQueryParameters) AddTimespan(key string, value time.Duration) *StatementQueryParameters {
+	return q.addBase(key, string(types.Timespan), newValue(value, types.Timespan))
+}
+
+func (q *StatementQueryParameters) AddDecimal(key string, value value.Decimal) *StatementQueryParameters {
```

Comment:
Why did you change it from decimal.decimal? now creating the value is much more awkward

Reply:
fitted it by a similar test. i can change it back i believe, but how is it less troublesome? ill still need 
kusto.ParamType{Type: types.Decimal, Default: "2.00000000000001"}
or something like that, no?

Reply:
decimal.Decimal is created by decimal.RequireFromString/decimal.NewFromString, which are much simpler, and expecting the user to past {Valid: true} is weird.
types.Decimal is for our internal conversion from kusto.